browser/net/brave_proxying_url_loader_factory.cc

```diff
@@ -351,44 +351,49 @@ void BraveProxyingURLLoaderFactory::InProgressRequest::
 
   // TODO(iefremov): Shorten
   if (ctx_->blocked_by != brave::kNotBlocked) {
-    if (!ctx_->ShouldMockRequest()) {
+    if (!ctx_->ShouldMockOrRedirectRequest()) {
       OnRequestError(
           network::URLLoaderCompletionStatus(net::ERR_BLOCKED_BY_CLIENT));
       return;
     }
+    // Check if it's a redirect URL or a data:// URL 
```

Comment:
it's not necessary to load the data like this, see `OnBeforeURLRequest_CommonStaticRedirectWork`

Reply:
@bridiver I moved the check to `OnBeforeURLRequest_CommonStaticRedirectWork`